app/core/src/main/java/com/fsck/k9/helper/ListUnsubscribeHelper.kt

```diff
@@ -0,0 +1,63 @@
+package com.fsck.k9.helper
+
+import android.net.Uri
+import com.fsck.k9.mail.Message
+import java.util.regex.Pattern
+
+class ListUnsubscribeHelper {
+    companion object {
+        private const val LIST_UNSUBSCRIBE_HEADER = "List-Unsubscribe"
+        private val MAILTO_CONTAINER_PATTERN = Pattern.compile("<(mailto:.+)>")
+        private val HTTPS_CONTAINER_PATTERN = Pattern.compile("<(https:.+)>")
+
+        // As K-9 Mail is an email client, we prefer a mailto: unsubscribe method
+        // but if none is found, a https URL is acceptable too
+        fun getPreferredListUnsubscribeUri(message: Message?): UnsubscribeUri? {
+            if (message == null) {
+                return null
+            }
```

Comment:
This function is never called with `message` being `null`. I suggest changing the parameter's type to `Message` and getting rid of this check.